### PR TITLE
github actions: don't use tj-actions/changed-files

### PR DESCRIPTION
**Pull request by ericLemanissier**
My motivation for not using `tj-actions/changed-files` actions any more is that:
- it breaks its interface (parameters) often, most release are major releases
- it documents no migration path between major versions
- breaking changes are not clearly identified

And we cannot stay on `tj-actions/changed-files/v20` for long because it uses deprecated features which will be removed soon (cf https://github.com/conan-io/conan-center-index/actions/runs/3326299661/jobs/5499882522#step:4:99 https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/)

This PR demonstrates that for CCI's use case (getting files changed in a pull-request), [7 lines of python code](https://github.com/conan-io/conan-center-index/pull/13748/files#diff-386b4a517af2610b191a5c1b415c1706c1ccda359d47ae624fa542df781c50adR32) are sufficient to do the job, so it feels to me that it's a smaller maintenance burden to use our custom action than tj-actions/changed-files

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by prince-chrismc** [comment]
I'll add this to our weekly meeting so we can move on this more quickly

**Comment by ericLemanissier** [comment]
thanks. There is not much hurry for this. I just updated the PR description with a better rationale for the change.

**Comment by ericLemanissier** [comment]
@prince-chrismc just to be clear, this PR is "ready for review", but probably not "ready to be merged"

**Review by danimtb — APPROVED** [review]
I think it is looking good! let's keep the recipe changes for other PRs and move it forward

File: .github/actions/pr_changed_files/action.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - uses: actions/setup-python@v4
+    - name: Get changed files
+      id: changed-files
+      shell: python
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      run: |
+        import json
+        import subprocess
+        import fnmatch
```
Comment (prince-chrismc) [inline .github/actions/pr_changed_files/action.yml:29]: Are they standard python nodules?

**Comment by danimtb** [comment]
> I'd like to propose a test.
> 
> I want to merge this into my repos master branch. Then have someone without writing access to open a few PRs - We can steal real from from CCI - and make sure we get the correct results.
> 
> This is the closest I can think of for a real world test.
> 
> Thoughts?

ok, let's do that

**Comment by prince-chrismc** [comment]
https://github.com/prince-chrismc/conan-center-index/commit/46bbd46b2dac0a2c87e9ff0110d8c6b579e463f8

You should be able to make PRs against my repo :)

also random changes like docs, configs, linters -- this should be a good test 🤞

**Comment by prince-chrismc** [comment]
Opened https://github.com/prince-chrismc/conan-center-index/pull/44 so far so good!

***

I have one small quibble about the order of calling `setup-python` twice

- we can remove the one in the action assume its there (which it is by default on the runners)
- fix the order in the workflow (but we need to make sure to get it right and thats error prone)


image

**Comment by ericLemanissier** [comment]
I don't think it's worth the risk, considering it literally takes 0 seconds to install python.
There is another problem though: only one file of each type is processed,  even if each of my PRs modify two of each. I'll take a look later today

**Comment by ericLemanissier** [comment]
it looks ok now. see how https://github.com/conan-io/conan-center-index/actions/runs/3487128109/jobs/5834387846 did not lint any file (because it only lints recipes), but https://github.com/conan-io/conan-center-index/actions/runs/3487128109/jobs/5834387983 linted the expected test recipe.

Can you pull this change in your fork @prince-chrismc ?

**Comment by prince-chrismc** [comment]
changes are in and I triggered the PRs again 👍 

Maybe next week we can get on a Zoom for an hr and just knock this out? it will be slow going back and forth

**Comment by prince-chrismc** [comment]
The two PRs look really promising. They ran only every necessary so :+1:


I'd like to see a docs PR (aka no files changes) and a linter PR just to make sure those two workflows still work 🙏 than I think we can green lite this to go in

**Comment by ericLemanissier** [comment]
ok, see:
- https://github.com/prince-chrismc/conan-center-index/pull/47
- https://github.com/prince-chrismc/conan-center-index/pull/48

**Review by prince-chrismc — APPROVED** [review]
Everything checks out and looks to be running correctly from a fork 🤞

**Review by uilianries — APPROVED** [review]
LGTM

**Pull request merged**